src/components/offers/createForm/OfferForm.js

```diff
@@ -13,39 +14,64 @@ const useStyles = makeStyles((theme) => ({
 }))
 
 export const OfferForm = (props) => {
+    const { id } = useParams()
+
+
     const defaultFormState = {
-        name:"",
-        position:"",
-        localization:"",
-        minSalary:"",
-        maxSalary:"",
-        description:"",
-        creatorId:"",
-        skills:[]
+        name: "",
+        position: "",
+        localization: "",
+        minSalary: "",
+        maxSalary: "",
+        description: "",
+        creatorId: "",
+        skills: []
     }
+
+
+
     const {handleSubmit, watch, trigger, control, reset} = useForm({mode: "onChange", defaultValues:defaultFormState})
 
     const minSalary = watch("minSalary", null)
     const maxSalary = watch("maxSalary", null)
 
+    function updateOffer(data) {
+        defaultFormState.name = data.name
+        defaultFormState.position = data.position
+        defaultFormState.localization = data.localization
+        defaultFormState.minSalary = data.minSalary
+        defaultFormState.maxSalary = data.maxSalary
+        defaultFormState.description = data.description
+        defaultFormState.skills = data.skills
+        reset(defaultFormState)
```

Comment:
Ogólnie żeby nie używać tego reseta (chyba o to na DSie pytałeś), mógłbyś zrobić tak, że defaultFormState wyniesiesz do useState'a. Nie jestem pewny z drugiej strony, czy warte tak się pierdzielić, zwłaszcza że tak działa i nie wydaje się to jakieś bardzo złe (unless useForm bazuje na useState z defaultFormState, wtedy mogłoby być już trochę gorzej, bo teoretycznie nie powinno się modyfikować stanu). Anyway... @Pinioo wypowiedz się, czy takie używanie reseta jest sprzeczne z zasadami, bo ja na tej libce się nie znam.

Reply:
Co do tej libki to ja już nie wiem co tu jest zgodne, a co sprzeczne z zasadami, ale generalnie dopóki ten formularz się jakoś dynamicznie nie zmienia, to ten reset może być jakkolwiek używany póki działa.